README.md

```diff
@@ -22,173 +18,177 @@ Currently, our SDK supports interactions with our application-specific rollup ba
+
+In order to get user signatures, applications need to [generate "signers"](#how-do-applications-generate-and-use-signers).
+
+#### What are transactions that update blockchain state?
+A common transaction type is the transfer of asset ownership from one user to another (ie. asset sale). These operations require users to sign (approve) them to prove that they are valid.
+
+#### What are operations that require authorization?
+
+These operations add to or update data in Immutable's databases and typically require the authorization of an object's owner (ie. a user creating a project on Immutable X).
+
+### How do applications generate and use signers?
+
+In order to generate a signature, a user’s private key is required. However, a user directly giving an application a private key is risky as private keys allow anyone in possession of them full control of an account.
+
+Instead, an app can get an interface to the user's account which is called a "signer". To do this a prompt is shown which will allow the user to connect with their wallet application (ie. mobile or browser wallet). Once connected the app can begin asking the user to sign transactions and messages that they can choose to approve or deny.
+
+There are two ways to get signers in your application:
+
+1. Use our [Wallet SDK](https://docs.x.immutable.com/sdk-docs/wallet-sdk-web/overview)
+2. Generate your own
+
+As Immutable X enables applications to execute signed transactions on both Ethereum (layer 1) and StarkEx (layer 2), signers are required for both these layers.
+
+### 1. Generate signers using the Wallet SDK
+
+The [Wallet SDK Web](https://docs.x.immutable.com/sdk-docs/wallet-sdk-web/overview) provides connections to Metamask and WalletConnect browser wallets.
+
+See [this guide](https://docs.x.immutable.com/sdk-docs/wallet-sdk-web/quickstart) for how to set this up.
+
+### 2. Generate your own signers
 
 ```ts
 import { AlchemyProvider } from '@ethersproject/providers';
 import { Wallet } from '@ethersproject/wallet';
-import {
-  WalletConnection,
-  generateStarkWallet,
-  BaseSigner,
-} from '@imtbl/core-sdk';
+import { Utils, ImmutableX, Config } from '@imtbl/core-sdk';
 
-// Setup a provider and a signer
+// Set up a provider
 const ethNetwork = 'ropsten'; // or mainnet;
-const privateKey = YOUR_PRIVATE_KEY;
 const provider = new AlchemyProvider(ethNetwork, YOUR_ALCHEMY_API_KEY);
 
-// Generate your own WalletConnection
-const generateWalletConnection = async (
-  provider: AlchemyProvider,
-): Promise<WalletConnection> => {
-  // L1 credentials
-  const ethSigner = Wallet.createRandom().connect(provider);
-
-  // L2 credentials
-  const starkWallet = await generateStarkSigner(ethSigner);
-
-  return {
-    l1Signer,
-    l2Signer,
-  };
-};
+// Create signers
+const ethWallet = new Wallet(YOUR_PRIVATE_ETH_KEY);
+const ethSigner = ethWallet.connect(provider);
+const starkSigner = Utils.createStarkSigner(YOUR_PRIVATE_STARK_KEY);
```

Comment:
It's not clear where to get the Stark private key from? A guide or a link to Stark documentation would be helpful I think.

Reply:
Account for removed `Utils` namespace:

```suggestion
const starkSigner = createStarkSigner(YOUR_PRIVATE_STARK_KEY);
```